### PR TITLE
 Blockified Order Confirmation

### LINKED ISSUE #10021: Blockify the Order Confirmation Template
This epic tracks the conversion of the order confirmation template from a legacy block, to individual inner blocks.
```[tasklist]
### Tasks
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10025
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10057
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10058
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10053
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10054
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10052
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10051
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10050
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10048
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10049
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/10055
```

### PR DESCRIPTION
**Pull request by mikejolley**
## What

This PR introduces new blocks for the individual parts of the order confirmation page. This includes blocks to display:

- The order status
- The order summary (ID, purchaser, date, total)
- Order details/totals
- Shipping and billing addresses

This replaces the legacy confirmation block which can be found under Appearance > Editor > Templates > Order Confirmation. The switch between legacy and new blocks is a manual process that involves editing the template and clicking the "transform into blocks" button.

Fixes #10021

## Why

This change allows merchants to control the appearance of the order confirmation page. It also modernises how the order confirmation page is rendered.

## Testing Instructions

Perform the following tests as a logged in admin user unless otherwise specified.

1. Go to Appearance > Editor > Templates > Order Confirmation
2. Convert the legacy order confirmation block to individual blocks using the "transform into blocks" button. Confirm the new blocks appear.
3. Focus on each new block and confirm there is an appropriate icon, title, description, and style controls for each. Play around with text colours, background colours, and any other options available to ensure changes apply in the editor.
4. Save changes.
6. Go to the store frontend, add some items to the cart, and place an order.
7. Confirm the confirmation page matches the styling you applied and all blocks are present. You can leave this tab open for testing.
8. Edit the order confirmation page again. Remove some blocks, for example headings.
9. Save changes, then refresh the confirmation page on the frontend. Confirm changes have applied.

#### Downloadable Products

1. Add a downloadable product to your cart and complete checkout.
2. The order confirmation page will be shown—no downloads will be visible. Leave this tab open.
3. In a new tab, go to store admin and edit the order you just placed. Change its status to "processing". This will grant access to downloads.
4. Go back to the confirmation page and refresh.
5. Confirm the download section is now visible.

#### Unauthenticated Views

1. Place an order as a logged in admin user. Leave the confirmation page open.
2. Copy the page address.
3. Open up a new incognito browser window, or better yet, using a different browser. Paste the URL from earlier.
4. Confirm no order details are visible—you should only see some text and a login link as follows:

Screenshot 2023-09-01 at 13 14 39

5. Close all tabs and log out from your store.
6. Place a new order as a guest. Confirm the order confirmation page shows some order details, but hides email address, payment method, and the full billing/shipping address. This is the limited view that guests can see.
7. Leave the confirmation page open. Copy the page address.
8. Open up a new incognito browser window, or better yet, using a different browser. Paste the URL from earlier.
9. You should still see details if within 10 minutes of placing the order.
10. Wait 10 minutes then refresh the page.
11. Confirm you see a message asking you to either login or fill out the form to see order details as follows:

Screenshot 2023-09-01 at 13 17 39

12. Enter the incorrect email. See a notice rejecting the request "We were unable to verify the email address you provided. Please try again.".
13. Enter the correct email. See limited order details.
14. Edit the URL of the page to remove the order key. For example, `checkout/order-received/1772/?key=wc_order_GAE4XxQ4k7E77` would become `checkout/order-received/1772`
15. You should no longer be able to provide your email and see order details. The key is required. You will see a generic message only.

Screenshot 2023-09-01 at 13 53 30

#### Developer tests

Install a 3rd party extension which adds something to the order confirmation page. Confirm that the extension renders it's content on the new order confirmation page. You can find examples on WooComerce.com. One I tried was "Share your purchase" which appends social sharing to the confirmation page.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

Screenshot 2023-09-01 at 13 53 50

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Feature: For users of block based themes, the order confirmation template can now be converted to individual blocks for greater control over layout, styling, and appearance. Go to Appearance > Editor > Templates > Order Confirmation and use the "transform into blocks" button to enable this functionality.

### CODE REVIEW

**Comment by mikejolley** [comment]
e2e tests are still needed, and a decision on https://github.com/woocommerce/woocommerce-blocks/issues/10050, but this feature branch is close to feature complete so I think we're ok to start the final review and approval process @tarhi-saad. Marking as ready to review.

**Comment by ralucaStan** [comment]
@nielslange & @alexflorisca I added you as well as reviewers for this branch that adds new functionality.

**Comment by alexflorisca** [comment]
Thanks, I'll review this tomorrow

**Comment by ralucaStan** [comment]
1st of all, excellent testing steps you have written Mike. They cover a lot of use cases and are written with clarity. Kudos on that.

I detected mostly nice-to-haves and also focused on the security of the Order confirmation page. I'll post my findings in separate comments. This is the 1st one.

1. For downloadable products the Billing Address appears in the center; It would be nice to have it appear on the left; so to remove the Column parent of the Shipping Address as well when there is no shipping required; On mobile it looks ok, probably because the 1st column has no height

I saw this by placing an order logged in as an Admin. 
Screenshot 2023-09-05 at 11 46 53

2. Clicking the Edit page on this sends the user to the checkout template; while edit site sends the user correctly. I know why this is happening; could we hide the Edit page button on this page? Also a nice to have that can be handled in a separate ticket
Screenshot 2023-09-05 at 12 03 13

**Comment by ralucaStan** [comment]
3. The email form can be submitted empty 

Screenshot 2023-09-05 at 14 54 56


4. I'd create a separate task to improve the look and feel of these two steps with design support. Outside of this project.


Screenshot 2023-09-05 at 14 54 56

Screenshot 2023-09-05 at 14 54 56

**Comment by ralucaStan** [comment]
@mikejolley what is the difference that enforces the email form? The fact that the order has been placed by a guest? There is a link to log in as well when the form appears; 
Screenshot 2023-09-05 at 14 54 56

**Comment by mikejolley** [comment]
> @mikejolley what is the difference that enforces the email form? The fact that the order has been placed by a guest? There is a link to log in as well when the form appears;

Yes its based on being a guest order. That doens't mean the user does not have an account though I guess. I think its fine to show both.

> I'd create a separate task to improve the look and feel of these two steps with design support. Outside of this project.

I raised the topic of editor views for these other forms and views. I think @pmcpinto is aware. For now this just inherits what core does (although I tweaked the wording). 

> The email form can be submitted empty

https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/79ff4bcffc0efc8e813cea273bb7d991ee7268f2

> For downloadable products the Billing Address appears in the center; It would be nice to have it appear on the left; so to remove the Column parent of the Shipping Address as well when there is no shipping required; On mobile it looks ok, probably because the 1st column has no height

It's not possible to remove the column because this is outside of the conditional block. It would need to be solved with CSS, or remove columns completely. 

> Clicking the Edit page on this sends the user to the checkout template; while edit site sends the user correctly. I know why this is happening; could we hide the Edit page button on this page? Also a nice to have that can be handled in a separate ticket

https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/0fa72ddcb484a75f73d160359c2e7a4050937135

File: assets/js/blocks/order-confirmation/downloads/edit.tsx
Comment (alexflorisca) [inline assets/js/blocks/order-confirmation/downloads/edit.tsx]: From what I can tell, we are immediately invoking a function here that takes certain props from the `blockProps.style` object and returns them. Why can't we just destruct the `blockProps.style` object here?

File: assets/js/blocks/order-confirmation/status/block.json
Comment (alexflorisca) [inline assets/js/blocks/order-confirmation/status/block.json:1]: Could we use a heading or paragraph block here instead of a custom block for the status?

File: src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php
Comment (alexflorisca) [inline src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php:83]: Should we use order-confirmation here and probably in other place to avoid confusion?

File: src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php
Comment (alexflorisca) [inline src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php:1]: I'm guessing some of the logic in this Block also lives in the controller for Order Received in WC Core. Wondering how we can make sure that this doesn't diverge too much.

**Comment by mikejolley** [comment]
> It would be nice to have it appear on the left; so to remove the Column parent of the Shipping Address as well when there is no shipping required; On mobile it looks ok, probably because the 1st column has no height

@ralucaStan What theme was that? For me the column is right aligned, but full width on mobile. 

Screenshot 2023-09-05 at 16 29 14

The empty column is still there.

**Comment by nielslange** [comment]
Thanks for working on this, @mikejolley. Similar to @ralucaStan, I'll add my findings along I go. I just finished reviewing the first test case and below you can see my findings.

---

Nitpick: The hover color of links within the `Heading` of the `Order Totals Section` inner block only affects the text, but not the underline.

Screenshot 2023-09-05 at 20 36 47

---

While it’s possible to apply a color to the links within the `Order Totals` inner block, the applied link color is only visible in the frontend. In the site editor, the links are not viewed as such.

Screenshot 2023-09-05 at 20 41 04

---

The horizontal margin works for all inner blocks except the `Order Status` and the `Order Summary` inner block.

Screenshot 2023-09-05 at 21 00 57

---

Tomorrow, I'll continue reviewing the remaining test cases as well as review the code changes.

**Comment by nielslange** [comment]
> @ralucaStan What theme was that? For me the column is right aligned, but full width on mobile.

@mikejolley I think what @ralucaStan meant was that when there's not a shipping address, for example when the order only contains virtual products, the billing address should appear in full-width. Currently, it's shown on the right-hand side, leaving an empty gap on the left-hand side of the desktop. Please correct me if I'm wrong, @ralucaStan.

**Comment by ralucaStan** [comment]
> @ralucaStan What theme was that? For me, the column is right aligned, but full width on mobile.

Twenty Twenty-Three is the theme I am using; 

> @mikejolley I think what @ralucaStan meant was that when there's not a shipping address, for example when the order only contains virtual products, the billing address should appear in full-width. 

I don't think the Billing address should appear full width, because there are column settings applied to it in the editor; I was just proposing to remove the 1st column altogether, and let the billing column be the first one on the left.

**Comment by nielslange** [comment]
> I don't think the Billing address should appear full width, because there are column settings applied to it in the editor; I was just proposing to remove the 1st column altogether, and let the billing column be the first one on the left.

In that case, I wonder if we should just flip the order of the billing and shipping address. That would address this issue, wouldn't it?

**Comment by mikejolley** [comment]
> I was just proposing to remove the 1st column altogether, and let the billing column be the first one on the left.

~@ralucaStan This won't be possible without removing columns completely. The column is a parent block for the address wrapper, its not inside the wrapper. Columns and Column rely on each other.~

~If we need this functionality, we'll need to replace columns/column with something else.~

Fixed in https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/c3ab14fc8d7ac52c349231c921c9bb12fe929e52

File: src/BlockTypes/OrderConfirmation/Status.php
Comment (alexflorisca) [inline src/BlockTypes/OrderConfirmation/Status.php]: nitpick: spelling mistake - change `to` to `so`

File: src/Utils/BlockTemplateUtils.php
Comment (alexflorisca) [inline src/Utils/BlockTemplateUtils.php]: nitpick: & billing?

**Review by alexflorisca — COMMENTED** [review]
Nice work @mikejolley, beast of a PR! The code looks good, left a few comments and questions inline, but nothing major. I tested using the instructions (well written!) and mostly verything worked as expect,  but I did find one potential bug and one nice to have

**Potential bug:**

When following your instructions for downloadable products, the links to the products aren't disabled. When clicking the link I get a file not found (probably nothing to do with this PR) and when I click the back button, the "downloads remaining" is set to 0

https://github.com/woocommerce/woocommerce-blocks/assets/3966773/af0b527e-797e-4309-82b6-418d02c8ddbf

**And a nice to have**

When the shipping address isn't displayed on the Order Confirmation page, the billing address is on the right, which seems a bit out of place. When there is only 1 address, it would be good to either have it on the left or have it span the entire width

File: assets/js/blocks/order-confirmation/downloads/edit.tsx
Comment (alexflorisca) [reply assets/js/blocks/order-confirmation/downloads/edit.tsx]: This was answered by @tarhi-saad in Slack (just preference) and agreed to change to destructuring instead

**Comment by tarunvijwani** [comment]
Great PR and amazing testing steps! 🎉 
Apart from the above comments, I noticed that adding a background color affects the size of the content. Not sure if this is a Gutenberg issue or something we need to address.

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/024dbc14-5dfa-4894-9e8e-62bddbe415d2


https://github.com/woocommerce/woocommerce-blocks/assets/11503784/cdc66ec7-e568-42ec-b665-1b048c5e00b7

**Comment by nielslange** [comment]
> When following your instructions for downloadable products, the links to the products aren't disabled. When clicking the link I get a file not found (probably nothing to do with this PR) and when I click the back button, the "downloads remaining" is set to 0

@alexflorisca When using the default WooCommerce products, the links of the album and the single point to a file on the WooCommerce server. If you want to simulate a download, you can go ahead and replace the file URL with a local file. You can find the corresponding section within the product editor. There, you can also adjust the download limit. In my case, the download limit was set to 1 by default. After clicking the download link, the download limit then changed to 0, which is expected. If you leave the download limit field empty, you can download the file unlimited times and the order confirmation page indicates this with the ∞ sign.

Screenshot 2023-09-06 at 11 30 33

**Comment by nielslange** [comment]
@mikejolley Step 5. of the testing instructions for Downloadable Products asks to confirm that downloads are disabled. In my case, they are not disabled after I refresh the page, as mentioned in step 4., but only after I click the download link and also only if there's a download limit defined (see also https://github.com/woocommerce/woocommerce-blocks/pull/10056#issuecomment-1707641524). I wonder if we should update the testing instructions to make this clear. 🤔

For example, when not defining a download limit, the downloads are never disabled, and I can download the same file unlimited:

File: assets/js/blocks/order-confirmation/status/style.scss
Comment (nielslange) [inline assets/js/blocks/order-confirmation/status/style.scss]: Nitpick:
```suggestion
a {
  margin-right: $gap;
  
  &.has-link-color {
    color: inherit;
  }
}
```

**Comment by mikejolley** [comment]
Thanks @alexflorisca 

> When following your instructions for downloadable products, the links to the products aren't disabled. When clicking the link I get a file not found (probably nothing to do with this PR) and when I click the back button, the "downloads remaining" is set to 0

I think this is potentially your test product data. Its working here, and this uses the same code as core. For reference, my test data links to `https://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2017/08/single.jpg` for album. You can confirm on the original confirmation page if worried; the links are the same on both versions.

> When the shipping address isn't displayed on the Order Confirmation page, the billing address is on the right, which seems a bit out of place. When there is only 1 address, it would be good to either have it on the left or have it span the entire width

Fixed in https://github.com/woocommerce/woocommerce-blocks/commit/c3ab14fc8d7ac52c349231c921c9bb12fe929e52

**Review by nielslange — COMMENTED** [review]
Apart from the feedback I shared via comments, I noticed that when making the inner blocks  `Order Status`, `Order Summary` or `Order Totals Section` `wide` or `full-width`, the corresponding CSS class appears twice in the frontend:

Screenshot 2023-09-06 at 16 23 04

File: assets/js/blocks/order-confirmation/downloads/edit.tsx
Comment (mikejolley) [reply assets/js/blocks/order-confirmation/downloads/edit.tsx]: https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/dabfe835e37c2ec48128026a87ce09eaa43317ba

File: assets/js/blocks/order-confirmation/status/block.json
Comment (mikejolley) [reply assets/js/blocks/order-confirmation/status/block.json:1]: No, the status block can return different content depending on context, and runs some PHP side filters on the content. Heading and Paragraph blocks are user defined.

File: src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php
Comment (mikejolley) [reply src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php:83]: This is a woo core defined query_var, so while it would be nice, renaming this is beyond the scope of the project.

File: src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php
Comment (mikejolley) [reply src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php:1]: Yeah; the email_verification_required part was pulled from core shortcodes. Thats not public so we're unable to reuse that, but we wanted to implement the same pattern. I am not sure there is a solution there without a core refactor but this type of problem can be worked on during the core merge and path to default. I would like to keep the scope down for this project.

File: src/BlockTypes/OrderConfirmation/Status.php
Comment (mikejolley) [reply src/BlockTypes/OrderConfirmation/Status.php]: https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/db7b644271e190071da53221ef1ce4aa14505d81

**Comment by mikejolley** [comment]
@tarunvijwani thanks,

> Apart from the above comments, I noticed that adding a background color affects the size of the content. Not sure if this is a Gutenberg issue or something we need to address.

This is intentional—if you define a background color, default padding is increased to compensate. You can still override padding in block settings. This pattern was suggested by design, and some of the core blocks also do this (paragraph).

File: src/Utils/BlockTemplateUtils.php
Comment (mikejolley) [reply src/Utils/BlockTemplateUtils.php]: https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/017ce1b9cb1758f270e7818e8822b77ae633df9c

File: assets/js/blocks/order-confirmation/status/style.scss
Comment (mikejolley) [reply assets/js/blocks/order-confirmation/status/style.scss]: 8288b381245ec49793df59b8bd4a391c1ef7355e

**Comment by alexflorisca** [comment]
> @alexflorisca When using the default WooCommerce products, the links of the album and the single point to a file on the WooCommerce server. If you want to simulate a download, you can go ahead and replace the file URL with a local file. You can find the corresponding section within the product editor. There, you can also adjust the download limit. In my case, the download limit was set to 1 by default. After clicking the download link, the download limit then changed to 0, which is expected. If you leave the download limit field empty, you can download the file unlimited times and the order confirmation page indicates this with the ∞ sign.

> I think this is potentially your test product data. Its working here, and this uses the same code as core. For reference, my test data links to https://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2017/08/single.jpg for album. You can confirm on the original confirmation page if worried; the links are the same on both versions.

Replying to both of your comments here @nielslange and @mikejolley. I'm not so worried about the empty file here, but I thought the links shouldn't be clickable at all? I guess I'm a bit confused by what "Confirm the downloads are disabled." means. Also if they should be disabled when the order status is "processing", when should they be enabled?

**Comment by mikejolley** [comment]
@alexflorisca I will reword the test. I think that should say "visible". Links are never disabled, when you have no access the section is hidden.

> @mikejolley Step 5. of the testing instructions for Downloadable Products asks to confirm that downloads are disabled. In my case, they are not disabled after I refresh the page, as mentioned in step 4.,

@nielslange updated instructions—it should read "visible"

**Comment by mikejolley** [comment]
@nielslange R/e your comments

> The horizontal margin works for all inner blocks except the Order Status and the Order Summary inner block.

I looked into this and it appears to be intentional by Gutenberg for "wide" width containers when the group is using a constrained layout. If it has a wide width, auto margins are automatically applied. This is not within our control and should not be fixed within the scope of the project—I imagine its a UX thing that needs resolving upstream (these apis are not final).

The margins work on the other blocks because they are inner blocks within a parent block. You can make margins work by selecting the parent group and changing the setting:

Screenshot 2023-09-06 at 11 08 24

What I did do is allow the parent level blocks (download wrapper, total wrapper) to have styles also if you want to control alignment/margins at that level instead for consistency. I also fixed the duplicate markup saved with the post.

https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/f67b9270f51d8c17d9ea4c6fb74d754e027b7bf1

> Nitpick: The hover color of links within the Heading of the Order Totals Section inner block only affects the text, but not the underline.

I will skip this one because Heading is a core block (`core/heading`).

> While it’s possible to apply a color to the links within the Order Totals inner block, the applied link color is only visible in the frontend. In the site editor, the links are not viewed as such.

Fixed in https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/952da92dc6a54f799d9678ef8905b40d60c20522

**Comment by mikejolley** [comment]
Thanks @nielslange @ralucaStan @alexflorisca @tarunvijwani for reviewing. I've pushed fixes for the feedback items that required changes.

Just noting here that there are the following items outstanding before this can be merged to trunk.

Extension support:
https://github.com/woocommerce/woocommerce-blocks/pull/10842

Test framework:
https://github.com/woocommerce/woocommerce-blocks/pull/10840

And finally e2e tests need to be written:
https://github.com/woocommerce/woocommerce-blocks/issues/10762

**Comment by mikejolley** [comment]
@ralucaStan https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/c7f3594096edff0acad0379fbe3d509e279e653e pushes the verification form outside of the parent block markup. I have not created a dedicated block for it.

Screenshot 2023-09-06 at 17 06 00

**Comment by ralucaStan** [comment]
> @ralucaStan [c7f3594](https://github.com/woocommerce/woocommerce-blocks/commit/c7f3594096edff0acad0379fbe3d509e279e653e) pushes the verification form outside of the parent block markup. I have not created a dedicated block for it.
> 
> Screenshot 2023-09-06 at 17 06 00

You were fast Mike. I appreciate your proactivity on this one. I'll test this change tomorrow.

File: src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php
Comment (ralucaStan) [reply src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php:1]: @mikejolley let's make an issue for your suggestion and make sure we don't forget about it;

**Comment by ralucaStan** [comment]
> Just noting here that there are the following items outstanding before this can be merged to trunk.

I've added the blocked label to the PR so that we can wait for the rest of the PRs

**Comment by ralucaStan** [comment]
@mikejolley I've tested with Gift Cards and this is how it looks on the template with blocks vs the old template.
I recommend disabling the list bullets and maybe adjusting the spacing. We can get design feedback as well.

It might be worth looking at product bundles and how those are rendered.

this branch
-----
Screenshot 2023-09-07 at 18 54 54

**another block theme**
------
Screenshot 2023-09-07 at 18 58 19

**shortcode**
-----
Screenshot 2023-09-07 at 19 00 25

**Comment by ralucaStan** [comment]
@mikejolley I no longer see the checkout form for an order placed by a guest in incognito. I do see the form after 10 minutes have expired and I refresh the original link. 

@nielslange or @tarunvijwani could somebody retest the Unauthenticated Views step?

**Comment by nielslange** [comment]
> I also fixed the duplicate markup saved with the post.

@mikejolley I still see `alignwide alignwide` for the following inner blocks:
- `woocommerce/order-confirmation-status`
- `woocommerce/order-confirmation-summary`
- `woocommerce/order-confirmation-totals-wrapper`

Screenshot 2023-09-11 at 12 35 59

Here's what I did:
- Check out the updated version.
- Run `nvm use && composer i && npm i && npm start`.

Do you have an explanation why I'm still seeing the alignment classes duplicated?

**Comment by nielslange** [comment]
> > Nitpick: The hover color of links within the Heading of the Order Totals Section inner block only affects the text, but not the underline.
> 
> I will skip this one because Heading is a core block (`core/heading`).

Sounds good, @mikejolley. I reported this issue on https://github.com/WordPress/gutenberg/issues/54340.

> > While it’s possible to apply a color to the links within the Order Totals inner block, the applied link color is only visible in the frontend. In the site editor, the links are not viewed as such.
> 
> Fixed in [952da92](https://github.com/woocommerce/woocommerce-blocks/commit/952da92dc6a54f799d9678ef8905b40d60c20522)

Thanks for fixing this. 🙌

**Comment by nielslange** [comment]
> @nielslange or @tarunvijwani could somebody retest the Unauthenticated Views step?

@mikejolley When I test the Unauthenticated Views test case, in step 9. I'm seeing the limited order details of the guest order when pasting the order link into a different browser window:

Screenshot 2023-09-11 at 12 56 22

- Left-hand side: Chrome → this is where I placed the guest order 
- Right-hand side: Firefox → this is where I opened the order confirmation page, I copied from Chrome

**Comment by mikejolley** [comment]
> I no longer see the checkout form for an order placed by a guest in incognito. I do see the form after 10 minutes have expired and I refresh the original link.

> @mikejolley When I test the Unauthenticated Views test case, in step 9. I'm seeing the limited order details of the guest order when pasting the order link into a different browser window:

@nielslange Ill update the instructions. There is a 10 minute window (now fixed) which allows access via the link. Once this expires, the form is then required.

**Comment by mikejolley** [comment]
> I still see alignwide alignwide for the following inner blocks

Fixed in https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/32e6afc9a77f082a72cd4dd48f919eb1409b5ef4

**Comment by mikejolley** [comment]
> I've tested with Gift Cards and this is how it looks on the template with blocks vs the old template.
I recommend disabling the list bullets and maybe adjusting the spacing. We can get design feedback as well.

@ralucaStan 

Fixed in https://github.com/woocommerce/woocommerce-blocks/pull/10056/commits/7ab297a61aa9168e6ee9dc6f5f49bcc451c70cae

Screenshot 2023-09-11 at 11 43 16

**Comment by tarhi-saad** [comment]
I bumped this PR to the next release!

**Comment by nielslange** [comment]
> @nielslange Ill update the instructions. There is a 10 minute window (now fixed) which allows access via the link. Once this expires, the form is then required.

Thanks for updating the testing steps accordingly, @mikejolley.

**Comment by nielslange** [comment]
> > I still see alignwide alignwide for the following inner blocks
> 
> Fixed in [32e6afc](https://github.com/woocommerce/woocommerce-blocks/commit/32e6afc9a77f082a72cd4dd48f919eb1409b5ef4)

Thanks for fixing this. The fix is now working as expected.

**Review by nielslange — APPROVED** [review]
Thanks for addressing all feedback, @mikejolley. I retested the PR again, and it works as expected on my end now. Good work. You have my approval already. 🙌

**Pull request merged**
